lib/services/iotaRedirector.js

```diff
@@ -310,27 +311,13 @@ function processRequests(req, res, next) {
                             ']'
                     )
                 );
-            } else if (body && body.length > 0) {
-                let parsedBody;
+            } else if (body) {
                 let parseError;
-
-                try {
-                    parsedBody = JSON.parse(body);
-                    if (response.statusCode === 409) {
-                        parseError = new errors.DuplicateError(parsedBody);
-                    }
-                    logger.debug(context, 'Response body from the redirected request parsed: \n\n%j\n', parsedBody);
-                } catch (e) {
```

Comment:
In this case the request-shim is expecting a JSON response. It already internally does a `JSON.parse()` If the response is not JSON `got` throws an error and the error is handled in https://github.com/jason-fox/iotagent-manager/blob/feature/got/lib/services/iotaRedirector.js#L286

Since the `got` library is already applying `JSON.parse(body);` internally a valid response is already a JSON object so https://github.com/jason-fox/iotagent-manager/blob/master/lib/services/iotaRedirector.js#L318  is unnecessary. There is no need to `JSON.parse()` within the IoT Manager code base and no need to add additional error handling  therefore the `try .. catch` is not needed.

Reply:
OK, NTC